server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/workspaceContext/semanticSearch.ts

```diff
@@ -0,0 +1,134 @@
+                contextParams: {
+                    documentContextParams: {
+                        query: params.query,
+                        queryConfigurationMap: {
+                            SEMANTIC: {
+                                maxResult: 15,
+                                includeDependencies: false,
+                                ...(params.programmingLanguage && { programmingLanguage: params.programmingLanguage }),
+                            },
+                        },
+                    },
+                },
+            },
+            {
+                headers: {
+                    Authorization: `Bearer ${creds.token}`,
+                },
+            }
+        )
+
+        return this.createOutput(response.data.contextResult?.documentContext?.queryOutputMap?.SEMANTIC)
+    }
+
+    private createOutput(semanticSearchResult: CodeChunkResult[] | undefined): InvokeOutput {
+        const filteredResults =
+            semanticSearchResult?.map(result => {
+                const normalizedUri = result.fileUri.startsWith('file://')
+                    ? result.fileUri
+                    : `file://${result.fileUri.startsWith('/') ? result.fileUri : '/' + result.fileUri}`
+
+                return {
+                    fileUri: normalizedUri,
+                    content: result.content,
+                    ...(result.score !== undefined && { similarityScore: result.score }),
+                }
+            }) || []
+
+        return {
+            output: {
+                kind: 'json',
+                content: filteredResults,
+            },
+        }
+    }
+
+    public getSpec() {
+        return {
+            name: SemanticSearch.toolName,
+            description:
+                'A tool for finding semantically relevant code snippets in a codebase.\n\n' +
+                '## Overview\n' +
+                'This is a semantic search tool that understands the intent and context behind queries, helping you find code snippets most relevant to your search.\n\n' +
+                '## When to use\n' +
+                '- When you need to locate specific functionality in a codebase\n' +
+                '- When looking for implementation patterns related to certain concepts\n' +
+                '- When you want to understand how particular features are coded\n' +
+                '- When exploring unfamiliar codebases to find relevant sections\n\n' +
+                '## When not to use\n' +
+                '- When you already know the exact file location\n\n' +
+                '## Notes\n' +
+                '- Before searching, identify the essential concepts and atomic information units in the query\n' +
+                '- For complex questions, break down the query into core components or key facts to improve search relevance\n' +
+                '- If it makes sense to only search in particular directories, specify them in the target_directories field\n' +
```

Comment:
target_directories does not exist?

Reply:
Let me confirm with science on this, agree this line seems to be extra